File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        if(n == 1)
            return 0;
        
        int res = 0;
        int l = 0;
        for(int r = 1; r < n; r++) {
            int temp = prices[r] - prices[l];
            if(temp < 0)
                l = r;
            else
                res = max(res, temp);
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        res = 0
        for i in nums:
            if i - 1 in s:
                continue
            else:
                temp = 1
                j = i + 1
                while j in s:
                    j += 1
                    temp += 1
                res = max(res, temp)
        return res